stdlib/public/core/Flatten.swift.gyb

```diff
@@ -250,7 +173,20 @@ public func < <BaseElements> (
   lhs: ${Index}<BaseElements>, 
   rhs: ${Index}<BaseElements>
 ) -> Bool {
-  fatalError("FIXME: swift-3-indexing-model implement")
+  if lhs._outer != rhs._outer {
+    return lhs._outer < rhs._outer
+  }
+
+  if let lhsInner = lhs._inner, rhsInner = rhs._inner {
+    return lhsInner < rhsInner
+  }
+
+  // When combined, the two conditions above guarantee that both
+  // `_outer` indices are `_base.endIndex` and both `_inner` indices
+  // are `nil`, since `_inner` is `nil` iff `_outer == base.endIndex`.
+  _precondition(lhs._inner == nil && rhs._inner == nil)
```

Comment:
I'm probably misunderstanding something, but why can't it happen so that in `lhs._inner == nil` and `rhs._inner != nil`?  It is valid to compare with `endIndex`.